src/main/java/org/broadinstitute/hellbender/tools/exome/DiscoverCopyNumberTriStateSegments.java

```diff
@@ -0,0 +1,484 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import com.google.common.collect.Iterators;
+import org.broadinstitute.hellbender.cmdline.*;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.exome.hmm.CopyNumberTriStateHiddenMarkovModel;
+import org.broadinstitute.hellbender.tools.exome.hmm.CopyNumberTriStateHiddenMarkovModelArgumentCollection;
+import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.hmm.CopyNumberTriState;
+import org.broadinstitute.hellbender.utils.hmm.ForwardBackwardAlgorithm;
+import org.broadinstitute.hellbender.utils.hmm.ViterbiAlgorithm;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+
+/**
+ * Discover potential copy-number segments in input read-counts as the most likely segment sequences
```

Comment:
Javadoc should indicate what step in workflow immediately precedes this (tangent normalization?)